rskj-core/src/main/java/co/rsk/peg/PegUtils.java

```diff
@@ -0,0 +1,38 @@
+package co.rsk.peg;
+
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.core.TransactionOutput;
+import co.rsk.bitcoinj.wallet.Wallet;
+import co.rsk.peg.bitcoin.BitcoinUtils;
+import org.ethereum.config.blockchain.upgrades.ActivationConfig;
+import org.ethereum.config.blockchain.upgrades.ConsensusRule;
+
+import java.util.List;
+import java.util.Optional;
+
+public class PegUtils {
+    private PegUtils() { }
+
+    public static PegTxType getTransactionType(
+        ActivationConfig.ForBlock activations,
+        BridgeStorageProvider provider,
+        Wallet liveFederationsWallet,
+        BtcTransaction btcTransaction
+    ) throws RegisterBtcTransactionException {
+        if (!activations.isActive(ConsensusRule.RSKIP379)){
+            throw new RegisterBtcTransactionException("Can't call PegUtil.getTransactionType before RSKIP379 activation.");
```

Comment:
Is this the right exception to throw here? Seems specific to a certain method and this in theory a utils method that could be used by anyone

Reply:
Makes sense. I will return an `illegalStateException` then.

Reply:
Done